### DISCUSSION TITLE
CloudInit logs

### DISCUSSION THREAD

**Discussion by cristian98149** [kairos-io/kairos#1164] [Q&A] [answered]
Hi,
is there a  way to view logs about errors encauntered during cloud-init execution?

For example, if a something in cloud-init file (for example a command) fails, how can we get logs about this?

Thank you

**Comment by Itxaka** [comment, marked as answer]
Hi!

on version lower than 2.0.0, the "cloud-init" stages execution is done via systemd services so you should be able to gather their logs with a simple journalctl -u cos-setup-STAGE

The service names are as follows:

cos-setup-rootfs
cos-setup-initramfs
cos-setup-network
cos-setup-boot
cos-setup-fs
cos-setup-reconcile


For versions higher or equal to 2.0.0, the rootfs and initramfs stages are now run via [immucore](https://github.com/kairos-io/immucore) and that stores the rootfs and initramfs logs to the /run/immucore dir.

 It also allows you to enable debug mode for the runs by settings `rd.immucore.debug` in the cmdline
